models/amity.py

```diff
@@ -0,0 +1,428 @@
+import random
+from rooms import Office, Living
+from people import Staff, Fellow, StaffFromDatabase, FellowFromDatabase
+from termcolor import colored
+
+spacer = " "
+border = colored("*" * 5, 'cyan').center(80)
+
+
+class Amity(object):
+    def __init__(self):
+        self.rooms = []
+        self.vacant_rooms = []
+        self.offices = []
+        self.vacant_offices = []
+        self.livingspaces = []
+        self.vacant_livingspaces = []
+        self.people = []
+        self.allocated_people = []
+        self.unallocated_people = []
+        self.fellows = []
+        self.allocated_fellows = []
+        self.staff = []
+        self.allocated_staff = []
+
+    def create_room(self, args):
+        """Create new room(s)"""
+        print spacer
+        new_rooms = []
+        for room in args["<room_name>"]:
+            if room.lower() in [r.name.lower() for r in self.rooms]:
+                print "One or more rooms you tried to create already exist! " \
+                    "Please try again."
+                print spacer
+                return
+            if args["Office"]:
+                new_room = Office(room)
+                self.offices.append(new_room)
+                self.check_vacant_offices()
+            elif args["Living"]:
+                new_room = Living(room)
+                self.livingspaces.append(new_room)
+                self.check_vacant_livingspaces()
+            self.rooms.append(new_room)
+            new_rooms.append(new_room)
+        print "You have successfully added the following rooom(s):"
+        for new_room_ in new_rooms:
+            print "Name: " + ''.join(new_room_.name) + " | Type: " \
+                + new_room_.room_type
+        print spacer
+
+    def create_room_from_db(self, args):
+        """Create new room(s) from existing database"""
+        print spacer
+        new_rooms = []
+        for room in args["<room_name>"]:
+            if room.lower() in [r.name.lower() for r in self.rooms]:
+                print "One or more rooms you tried to create already exist! " \
+                    "Please try again."
+                print spacer
+                return
+            if args["Office"]:
+                new_room = Office(room)
+                self.offices.append(new_room)
+                self.check_vacant_offices()
+            elif args["Living"]:
+                new_room = Living(room)
+                self.livingspaces.append(new_room)
+                self.check_vacant_livingspaces()
+            self.rooms.append(new_room)
+            new_rooms.append(new_room)
+        print "You have successfully added the following rooom(s):"
+        for new_room_ in new_rooms:
+            print "Name: " + ''.join(new_room_.name) + " | Type: " \
+                + new_room_.room_type
+        print spacer
+
+    def check_vacant_offices(self):
```

Comment:
It's required to reduce redundancy. There are several instances where I need to check whether a room is vacant, e.g when adding a person who wants space, or when reallocating a person another room. So rather than re-write the same piece of code I opted to write a method that I could call as required.